src/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -63,8 +67,10 @@
 	private ContentletFactory conFac = FactoryLocator.getContentletFactory();
 	private HostCache hostCache = CacheLocator.getHostCache();
 	private Host systemHost;
+	private final SystemEventsAPI systemEventsAPI;
 
 	public HostAPIImpl() {
+		this.systemEventsAPI = APILocator.getSystemEventsAPI();

```

Comment:
This should be supplied by another constructor, following the same approach we follow for other classes.